classes/proxy_lock_factory.php

```diff
@@ -464,4 +458,24 @@ private function fill_more_for_tasks($record, $resourcekey) {
         return $record;
     }
 
-}
+    /**
+     * Get lock type.
+     * @param stdClass $record
+     * @return stdClass $record
+     */
+    private function get_lock_type($record) {
```

Comment:
this looks good, but I find the name misleading as it is not a getter, it updated the record in situ, perhaps update_lock_type() ?

Reply:
also as you are mutating the record in place, I'd return null?